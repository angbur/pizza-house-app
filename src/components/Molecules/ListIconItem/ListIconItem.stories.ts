import { Meta, StoryObj } from '@storybook/react';
import ListIconItem from './ListIconItem';

const meta = {
  title: 'Molecules/IconItem',
  component: ListIconItem,
  tags: ['docsPage'],
} satisfies Meta<typeof ListIconItem>;

export default meta;

type Story = StoryObj<typeof meta>;

export const MainIconItem: Story = {
  args: {
    label: 'order',
    icon: {
      name: 'order',
      size: 32,
    },
  },
};
