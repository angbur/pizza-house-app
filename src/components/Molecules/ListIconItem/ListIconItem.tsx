import { CSSProperties, PropsWithChildren } from 'react';
import styled from 'styled-components';
import Typography from 'components/Atoms/Typography/Typography';
import Icon, { IconName } from 'components/Atoms/Icon/Icon';

export type ListIconItemProps = {
  label: string;
  icon: {
    name: IconName;
    size: number;
  };
  isActive: boolean;
  style?: CSSProperties;
};

const IconItemBox = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 10px;
  padding: 1.25rem 0;
  cursor: pointer;
`;

const ListIconItem = ({
  label,
  icon,
  style,
  isActive = false,
}: PropsWithChildren<ListIconItemProps>) => (
  <li>
    <IconItemBox>
      {Icon({ ...icon, isActive })}
      <Typography
        variant={'item-title'}
        color='light'
        style={{ ...{ textTransform: 'uppercase' }, ...style }}
      >
        {label}
      </Typography>
    </IconItemBox>
  </li>
);

export default ListIconItem;
