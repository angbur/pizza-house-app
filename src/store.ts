import { configureStore } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import { pizzaApi } from 'services/pizza';

export const store = configureStore({
  reducer: {
    order: orderReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pizzaApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
