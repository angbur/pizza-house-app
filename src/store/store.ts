import { configureStore, Store } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import dialogReducer from 'components/Organisms/Dialog/dialogSlice';
import { pizzaApi } from 'services/pizza';
import orderMiddleware from './middleware/orderMiddleware';

export const store: Store = configureStore({
  reducer: {
    order: orderReducer,
    dialog: dialogReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pizzaApi.middleware, orderMiddleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
