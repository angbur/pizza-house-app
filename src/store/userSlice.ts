import { createSlice } from '@reduxjs/toolkit';
import { RootState } from './store';
import { toast } from 'react-toastify';
import { userApi } from 'services/user';

type User = {
  loggedIn: boolean;
  status: string | undefined;
  error: string | undefined;
  userId: string | undefined;
  login: string | undefined;
  token: string | undefined;
};

const initialState: User = {
  loggedIn: false,
  status: 'idle',
  error: undefined,
  userId: undefined,
  login: undefined,
  token: undefined,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logout: (state) => {
      state.loggedIn = false;
      state.userId = undefined;
      state.login = undefined;
      state.token = undefined;
      toast.success('Logged out!');
    },
  },
  extraReducers: (builder) => {
    builder.addMatcher(userApi.endpoints.login.matchFulfilled, (state, { payload }) => {
      state.loggedIn = true;
      state.token = payload.token;
      state.userId = payload.userId;
      state.login = payload.login;
    });
  },
});

export const { logout } = userSlice.actions;

export const selectName = (state: RootState) => state.user.user;

export const isLoggedIn = (state: RootState) => state.user.loggedIn;

export const selectResponseStatus = (state: RootState) => state.user.status;

export const selectResponseError = (state: RootState) => state.user.error;

export const selectLoggedInUser = (state: RootState) => state.user.user;

export default userSlice.reducer;
