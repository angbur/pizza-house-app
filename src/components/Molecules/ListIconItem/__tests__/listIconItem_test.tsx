import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import { store } from 'store/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import ListIconItem from '../ListIconItem';
import React from 'react';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <BrowserRouter>
        <Provider store={store}>
          <ListIconItem icon={{ name: 'order', size: 26 }} label={'order'} linkTo={'order'} />
        </Provider>
      </BrowserRouter>
    </ThemeProvider>
  );
};

describe('Order List Item', () => {
  test('should renders order list item correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      <li>
        <a
          className=""
          href="/order"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-bcXHqe gIzvfo icon-item"
          >
            <img
              height={26}
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "order.svg",
                }
              }
            />
            <h5
              className="item-title"
              style={
                Object {
                  "color": "none",
                  "fontFamily": "Source Sans Pro",
                  "fontSize": "1rem",
                  "fontWeight": "600",
                  "letterSpacing": "1px",
                  "lineHeight": "1.4",
                  "textTransform": "uppercase",
                }
              }
            >
              order
            </h5>
          </div>
        </a>
      </li>
    `);
  });
});
