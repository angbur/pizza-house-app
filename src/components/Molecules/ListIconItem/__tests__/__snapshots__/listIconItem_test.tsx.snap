// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order List Item should renders order list item correctly 1`] = `
<li>
  <a
    className=""
    href="/order"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="sc-bcXHqe gIzvfo icon-item"
    >
      <img
        height={26}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "order.svg",
          }
        }
      />
      <h5
        className="item-title"
        style={
          Object {
            "color": "none",
            "fontFamily": "Source Sans Pro",
            "fontSize": "1rem",
            "fontWeight": "600",
            "letterSpacing": "1px",
            "lineHeight": "1.4",
            "textTransform": "uppercase",
          }
        }
      >
        order
      </h5>
    </div>
  </a>
</li>
`;
